pkg/controller/node/node_controller.go

```diff
@@ -164,6 +167,36 @@ func filterSegmentPorts(nsxClients *NsxClients, ports []*data.StructValue, nodeN
 	return -1, errors.Errorf("No port matches")
 }
 
+func getConnectorTLSConfig(insecure bool, clientCertFile string, clientKeyFile string, caFile string) (*tls.Config, error) {
+	tlsConfig := tls.Config{InsecureSkipVerify: insecure}
+
+	if len(clientCertFile) > 0 {
+		if len(clientKeyFile) == 0 {
+			return nil, fmt.Errorf("Please provide key file for client certificate")
+		}
+
+		cert, err := tls.LoadX509KeyPair(clientCertFile, clientKeyFile)
```

Comment:
I found there is another function  X509KeyPair to use PEM encoded data directly, maybe we don't need to save to a file in the middle?